open_city_profile/views.py

```diff
@@ -0,0 +1,66 @@
+import sentry_sdk
+from django.core.exceptions import ObjectDoesNotExist, PermissionDenied
+from graphene_django.views import GraphQLView as BaseGraphQLView
+
+from open_city_profile.consts import (
+    GENERAL_ERROR,
+    OBJECT_DOES_NOT_EXIST_ERROR,
+    PERMISSION_DENIED_ERROR,
+    SERVICE_ALREADY_EXIST_ERROR,
+)
+from open_city_profile.exceptions import ProfileGraphQLError, ServiceAlreadyExistError
+
+error_codes_shared = {
+    Exception: GENERAL_ERROR,
+    PermissionDenied: PERMISSION_DENIED_ERROR,
+    ObjectDoesNotExist: OBJECT_DOES_NOT_EXIST_ERROR,
+}
+error_codes_profile = {ServiceAlreadyExistError: SERVICE_ALREADY_EXIST_ERROR}
+error_codes = {**error_codes_shared, **error_codes_profile}
+
+
+class GraphQLView(BaseGraphQLView):
+    def execute_graphql_request(self, request, data, query, *args, **kwargs):
+        """Extract any exceptions and send some of them to Sentry"""
+        result = super().execute_graphql_request(request, data, query, *args, **kwargs)
+        # If 'invalid' is set, it's a bad request
+        if result and result.errors and not result.invalid:
+            errors = [
+                e
+                for e in result.errors
+                if not isinstance(
+                    getattr(e, "original_error", None), ProfileGraphQLError
+                )
+            ]
+            if errors:
+                self._capture_sentry_exceptions(result.errors, query)
+        return result
+
+    def _capture_sentry_exceptions(self, errors, query):
+        with sentry_sdk.configure_scope() as scope:
+            scope.set_extra("graphql_query", query)
+            for error in errors:
+                if hasattr(error, "original_error"):
+                    error = error.original_error
+                sentry_sdk.capture_exception(error)
+
+    @staticmethod
+    def format_error(error):
+        def get_error_code(exception):
+            """Get the most specific error code for the exception via superclass"""
+            for exception in exception.mro():
```

Comment:
not bad 👍

Reply:
With heavy heart I threw recursion away 😄

Reply:
😭 😢 😿 👋 🚶